conf/modules.config

```diff
@@ -17,173 +17,296 @@ process {
+    // QUAST: Prefer to keep the local module since it can deal with the inputs I have
+    withName: QUAST {
+        ext.prefix = { "${meta.id}_${task.process.tokenize(':')[-3].toLowerCase()}" }
         publishDir = [
-            path: { "$params.outdir/assembly_evaluation/merqury/$meta.build" },
+            path: { "${params.outdir}/QC/${task.process.tokenize(':')[2].toLowerCase()}${task.process.tokenize(':')[2].toLowerCase() == 'assemble' ? '' : '/' + task.process.tokenize(':')[-3].tokenize('_')[1].toLowerCase() }/quast/${meta.id}" },
             mode: params.publish_dir_mode,
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }            
+        ]
+    }
+    // QUAST related mapping
+    // these are all local
+    withName: '.*MAP_TO_REF.*' {
+        publishDir = [
+            path: { "${params.outdir}/QC/${task.process.tokenize(':')[2].toLowerCase()}${task.process.tokenize(':')[2].toLowerCase() == 'assemble' ? '' : '/' + task.process.tokenize(':')[-3].tokenize('_')[1].toLowerCase() }/quast/${meta.id}/map_to_ref/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }            
+        ]
+    }
+    // QUAST related mapping
+    withName: '.*MAP_TO_ASSEMBLY.*' {
+        ext.prefix = { "${meta.id}-${task.process.tokenize(':')[2].toLowerCase()}" }
+        publishDir = [
+            path: { "${params.outdir}/QC/${task.process.tokenize(':')[2].toLowerCase()}${task.process.tokenize(':')[2].toLowerCase() == 'assemble' ? '' : '/' + task.process.tokenize(':')[-3].tokenize('_')[1].toLowerCase() }/quast/${meta.id}/map_to_assembly/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }            
+        ]
+    }
+    // MERQURY: nf-core 
+    withName: MERQURY {
+        ext.prefix = { "${meta.id}-${task.process.tokenize(':')[-3].toLowerCase()}" }
+        publishDir = [
+            path: { "${params.outdir}/QC/${task.process.tokenize(':')[2].toLowerCase()}${task.process.tokenize(':')[2].toLowerCase() == 'assemble' ? '' : '/' + task.process.tokenize(':')[-3].tokenize('_')[1].toLowerCase() }/merqury/${meta.id}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }            
         ]
     }
 
-    withName: 'MERQURYFK_MERQURYFK' {
-        tag        = { meta.build }
-        ext.prefix = { meta.build }
+    /* 
+    ----------
+    Polishing
+    ----------
+    */
+    // Medaka: local; nf-core module cant deal with gzipped input
+    withName: MEDAKA {
+        ext.args = [params.medaka_model ? "--model ${params.medaka_model}" : '' ]
```

Comment:
A general guideline is that any `ext.*` directive should be a closure if it uses a variable. https://nf-co.re/docs/guidelines/components/modules#optional-command-arguments